### PR TITLE
Added RestConnector.cmake

### PR DESCRIPTION
**Pull request by chevtche**

### CODE REVIEW

**Comment by chevtche** [comment]
First push for RestConnector

File: RestConnector.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+set(RESTCONNECTOR_VERSION 0.1)
+set(RESTCONNECTOR_REPO_URL https://github.com/BlueBrain/RESTConnector.git)
+set(RESTCONNECTOR_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/src/RestConnector")
```
Comment (eile) [inline RestConnector.cmake]: Not needed - and src dir name should match repo name.

**Comment by eile** [comment]
+1

File: RestConnector.cmake [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+set(RESTCONNECTOR_DEPENDS REQUIRED Boost cppnetlib Lunchbox zeq FlatBuffers
```
Comment (tribal-tec) [inline RestConnector.cmake]: FlatBuffers is used directly?

**Comment by chevtche** [comment]
Updated. FlatBuffers is not used directly, but zeq expose some .h from FlatBuffers so one needs to explicitly mark it as a denpendency. Otherwise: compilation error!

**Comment by eile** [comment]
That comes in through the generated vocabulary. Either don't include these as defaults, or declare FlatBuffers as a dependent library in zeq.

**Comment by tribal-tec** [comment]
Hmm, we probably should then generate another header filer which has no flatbuffers dependency. It should be possible for our use case. The transient dependency is ugly, I didn't realize that :(

**Comment by tribal-tec** [comment]
Btw, we need a review for RESTConnector itself. It was pushed directly to the repo w/o any review.

**Comment by favreau** [comment]
I have the feeling that removing the flatbuffer dependency is a different story. This needs code changes in zeq, not in the RestConnector itself. I would suggest that we create a separate issue for that, so we can merge the pull request. Any objections?

File: RESTConnector.cmake [hunk 3]
```diff
@@ -0,0 +1,6 @@
+set(RESTCONNECTOR_VERSION 0.1)
+set(RESTCONNECTOR_REPO_URL https://github.com/BlueBrain/RESTConnector.git)
+set(RESTCONNECTOR_DEPENDS REQUIRED Boost cppnetlib Lunchbox zeq FlatBuffers
+                          OPTIONAL BBPDocumentation)
```
Comment (tribal-tec) [inline RESTConnector.cmake]: should be bluebrain

File: RESTConnector.cmake [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+set(RESTCONNECTOR_BOOST_COMPONENTS "unit_test_framework program_options system thread")
+set(RESTCONNECTOR_DEB_DEPENDS libboost-test-dev libboost-program-options-dev)
```
Comment (tribal-tec) [inline RESTConnector.cmake]: dep list does not match components list. Please verify which ones are needed.

**Comment by chevtche** [comment]
Yes, it was a mistake from me... I thought the review will be automatic
but apparently i need to always fork and pull request.

**Comment by tribal-tec** [comment]
For this review it's fine. Just address the comments.
For RESTConnector, we probably will have another iteration where the review happens. As I think everybody has direct push access w/o needing a fork + PR, caution is required :)

**Comment by chevtche** [comment]
Updated.

**Pull request merged**